### PR TITLE
Spark CRD + Operator v1beta2

### PR DESCRIPTION
**Pull request by akhurana001**
Move spark plugin to v1beta2 multi-version of Operator.

Non-backward compatible: If the cluster already has Spark v1beta1 CRD & SparkOperator deployed, these will need to be manually deleted before the new CRD/operator can be deployed. This will cause any existing spark tasks/jobs running on the cluster to fail.  Ref: lyft/flyte#573 for more details.

Dependent on https://github.com/lyft/flyteplugins/pull/134

Updated FlytePropeller to version 0.5.0

### CODE REVIEW

File: kustomize/base/operators/spark/kustomization.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 resources:
 - spark-operator.yaml
-- spark-dummy-crd.yaml
+- sparkapplications-crd.yaml
```
Comment (kumare3) [inline kustomize/base/operators/spark/kustomization.yaml:3]: we should probably add this to a CRD section

**Comment by kumare3** [comment]
can you look at kustomize's CRD settings and add everything to that?

File: kustomize/base/operators/spark/kustomization.yaml (hunk 1 above)
Comment (akhurana001) [reply kustomize/base/operators/spark/kustomization.yaml:3]: resolving as we decided to have this as a resource

**Review by kumare3 — APPROVED** [review]

**Comment by kumare3** [comment]
Can we update the description to say that, this is a backwards incompatible change if you are using Spark!?

**Review by kumare3 — APPROVED** [review]

**Pull request merged**